lib/avalon/batch/entry.rb

```diff
@@ -18,126 +18,150 @@
-
-    	def initialize(fields, files, opts, row, manifest)
-	  @fields = fields
-	  @files  = files
-	  @opts   = opts
-	  @row    = row
-	  @manifest = manifest
-	  @errors = ActiveModel::Errors.new(self)
-	  @files.each { |file| file[:file] = File.join(@manifest.package.dir, file[:file]) }
-        end
-
-        def media_object
-          @media_object ||= MediaObject.new(avalon_uploader: @manifest.package.user.user_key, 
-                                            collection: @manifest.package.collection).tap do |mo|
-            mo.workflow.origin = 'batch'
-            if Avalon::BibRetriever.configured? and fields[:bibliographic_id].present?
-              begin
-                mo.descMetadata.populate_from_catalog!(fields[:bibliographic_id].first, Array(fields[:bibliographic_id_label]).first)
-              rescue Exception => e
-                @errors.add(:bibliographic_id, e.message)
-              end
-              # Sometimes we want to override the bib import data for some of the
-              # required fields (particularly if import data doesn't pass validations)
-              mo.update_datastream(:descMetadata,
-                                   fields.slice(:language,
-                                                :topical_subject,
-                                                :genre))
-            else
-              mo.update_datastream(:descMetadata, fields.dup)
+      extend ActiveModel::Translation
+
+      attr_reader :fields, :files, :opts, :row, :errors, :manifest, :collection
+
+      def initialize(fields, files, opts, row, manifest)
+        @fields = fields
+        @files  = files
+        @opts   = opts
+        @row    = row
+        @manifest = manifest
+        @errors = ActiveModel::Errors.new(self)
+        @files.each { |file| file[:file] = File.join(@manifest.package.dir, file[:file]) }
+      end
+
+      def media_object
+        return @media_object if @media_object
+        @media_object = nil
+        if fields[:media_object].present?
+          begin
+            @media_object = MediaObject.find(fields[:media_object].first)
+          rescue ActiveFedora::ObjectNotFoundError => e
+            @errors.add(:media_object, e.message)
+            return nil
+          end
+        else
+          @media_object = MediaObject.new(avalon_uploader: @manifest.package.user.user_key, 
+                                          collection: @manifest.package.collection)
+        end
+        update_media_object
+        return @media_object
+      end
+
+      def update_media_object
+        @media_object.tap do |mo|
+          mo.workflow.origin = 'batch'
+          if Avalon::BibRetriever.configured? and fields[:bibliographic_id].present?
+            begin
+              mo.descMetadata.
+                populate_from_catalog!(fields[:bibliographic_id].first,
+                                       Array(fields[:bibliographic_id_label]).first)
+            rescue Exception => e
```

Comment:
Probably. I should mention that I didn't write this line of code or the other one you commented on. Unfortunately I fixed the bad tabs/spaces indenting in the previous commit in the PR, so it looks like the patch is a lot bigger than it is. The stuff I actually wrote is in the second commit here:

 https://github.com/ualbertalib/avalon/pull/130/commits/07e8c4770d320f2a3314448487bff31eb83f71d1